content/ADR-164-quests.poc.md

```diff
@@ -0,0 +1,211 @@
+    loop
+        Quest Server -->> Explorer/Kernel: Send quest updates
+    end
+```
+
+- Real time updates
+
+Once the API is defined, the system may expose different ways to query quest states, one of them is to use some long living channel (let's say WebSockets).
+
+```mermaid
+sequenceDiagram
+    Explorer/Kernel ->> Quest Server: Send subscription request
+    loop
+        Quest Server -->> Explorer/Kernel: Send quest updates
+    end
+```
+
+### Workflows: Quest Designer 
+
+- Deploy new quest or modify existing quest
+
+When the quest definition is ready, the Quest Designer tool would be able to send the new or modified version to the Quest Server. The request should include the author signature.
+
+```mermaid
+sequenceDiagram
+    Quest Designer ->> Quest Designer: Quest Definition 
+    Quest Designer ->> Quest Server: Send Quest definition and signature
+    Quest Server ->> Quest Designer: Send Quest submission response 
+```
+
+- Retrieve quest stats
+
+The quest author may be interested in the stats, how many users started, completed, and other information like starting and ending times.
+
+```mermaid
+sequenceDiagram
+    Quest Designer ->> Quest Server: Query Quest state (include signature to validate authorship)
+    Quest Server ->> Quest Designer: Send Quest state 
+```
+
+## Specification
+
+About the implementation and alternatives, each actor has some important decisions to explore.
+
+### Communication between the System and Kernel
+
+In order to receive real time updates, there are two options to handle the communication:
+- Use a socket (peek any protocol like WebSocket, WebRTC, WebTransport, QUIC, etc) to keep posted on interested quests. 
+- Use a HTTP polling mechanism.
+
+### Quest Model persistence
+
+When a Quest definition is deployed to the System, it should be persisted in some form, the alternatives are:
+- Human Readable (JSON, toml, etc)
+- Binary
+No matter which format is, serialization/deserialization should be easy and fast and possible in any language.
+
+Also, the location of the persisted quests is important, for example:
+- Store in DB
+- Store as a File (S3 or any shared instance)
+
+### Quest Designer
+
+Designing a quest should be a good experience, without friction and intuitive.
+
+Asides from implementing a solution taking into account the possibility of visual edition and easy deployment, should also consider the accesibility of the interface i.e.:
+- Create a hosted service where designers can access as a WebApp.
+- Provide executables or guidelines to run the application.
+
+Authenticating designers may be required, so the alternatives are:
+- Use Decentraland Login and potentially add wallet validations to restrict deployments to land-owners or other conditions.
+- Use Service custom authentication and API tokens.
+
+### Backend
+
+To provide a robust system, it should be scalable, easy to deploy and provide metrics.
+
+Scalability must be thought from the beginning:
```

Comment:
What about centralization vs decentralization? And given the second, what about decentralization in terms of _architecture_ (i.e. sync of the whole quests system) vs _content_ (i.e. each quest designer having their own quests server)?